src/compose/images.ts

```diff
@@ -52,584 +47,598 @@ type NormalisedDockerImage = Docker.ImageInfo & {
-				const newImage = _.clone(image) as NormalisedDockerImage;
-				newImage.NormalisedRepoTags = await this.getNormalisedTags(image);
-				return newImage;
-			}),
-			db.models('image').select(),
-		]);
-		return cb(normalisedImages, dbImages);
-	}
+export async function getNormalisedTags(
+	image: Docker.ImageInfo,
+): Promise<string[]> {
+	return await Bluebird.map(
+		image.RepoTags != null ? image.RepoTags : [],
+		normalise,
+	);
+}
 
-	private addImageFailure(imageName: string, time = process.hrtime()) {
-		this.imageFetchLastFailureTime[imageName] = time;
-		this.imageFetchFailures[imageName] =
-			this.imageFetchFailures[imageName] != null
-				? this.imageFetchFailures[imageName] + 1
-				: 1;
-	}
+async function withImagesFromDockerAndDB<T>(
+	cb: (dockerImages: NormalisedDockerImage[], composeImages: Image[]) => T,
+) {
+	const [normalisedImages, dbImages] = await Promise.all([
+		Bluebird.map(docker.listImages({ digests: true }), async (image) => {
+			const newImage = _.clone(image) as NormalisedDockerImage;
+			newImage.NormalisedRepoTags = await getNormalisedTags(image);
+			return newImage;
+		}),
+		db.models('image').select(),
+	]);
+	return cb(normalisedImages, dbImages);
+}
 
-	private matchesTagOrDigest(
-		image: Image,
-		dockerImage: NormalisedDockerImage,
-	): boolean {
-		return (
-			_.includes(dockerImage.NormalisedRepoTags, image.name) ||
-			_.some(dockerImage.RepoDigests, (digest) =>
-				Images.hasSameDigest(image.name, digest),
-			)
-		);
-	}
+function addImageFailure(imageName: string, time = process.hrtime()) {
+	imageFetchLastFailureTime[imageName] = time;
+	imageFetchFailures[imageName] =
+		imageFetchFailures[imageName] != null
+			? imageFetchFailures[imageName] + 1
+			: 1;
+}
 
-	private isAvailableInDocker(
-		image: Image,
-		dockerImages: NormalisedDockerImage[],
-	): boolean {
-		return _.some(
-			dockerImages,
-			(dockerImage) =>
-				this.matchesTagOrDigest(image, dockerImage) ||
-				image.dockerImageId === dockerImage.Id,
-		);
-	}
+function matchesTagOrDigest(
+	image: Image,
+	dockerImage: NormalisedDockerImage,
+): boolean {
+	return (
+		_.includes(dockerImage.NormalisedRepoTags, image.name) ||
+		_.some(dockerImage.RepoDigests, (digest) =>
+			hasSameDigest(image.name, digest),
+		)
+	);
+}
 
-	public async getAvailable(): Promise<Image[]> {
-		const images = await this.withImagesFromDockerAndDB(
-			(dockerImages, supervisedImages) =>
-				_.filter(supervisedImages, (image) =>
-					this.isAvailableInDocker(image, dockerImages),
-				),
-		);
+function isAvailableInDocker(
+	image: Image,
+	dockerImages: NormalisedDockerImage[],
+): boolean {
+	return _.some(
+		dockerImages,
+		(dockerImage) =>
+			matchesTagOrDigest(image, dockerImage) ||
+			image.dockerImageId === dockerImage.Id,
+	);
+}
 
-		return images;
-	}
+export async function getAvailable(): Promise<Image[]> {
+	const images = await withImagesFromDockerAndDB(
```

Comment:
```suggestion
	return withImagesFromDockerAndDB(
```

... just a NIT but I think this is equivailent, no?